### PR TITLE
Warn pods not starting

### PR DESCRIPTION
**Pull request by nimrod-up9**
Print warning event related to mizu k8s resources.
In non-daemon print to CLI. In Daemon print to API-Server logs.

https://user-images.githubusercontent.com/59927337/142762097-b7074e09-e1bc-40b0-b01d-54c8915b9353.mov

### CODE REVIEW

**Review by gadotroee — APPROVED** [review]

File: cli/cmd/tapRunner.go [hunk 1]
```diff
@@ -736,6 +739,89 @@ func watchTapperPod(ctx context.Context, kubernetesProvider *kubernetes.Provider
...
+
+	for {
+		select {
+		case wEvent, ok := <-added:
+			if !ok {
+				added = nil
+				continue
+			}
+
+			event, err := wEvent.ToEvent()
```
Comment (RoyUP9) [inline cli/cmd/tapRunner.go:758]: maybe extract to function, i see the same code used 3 times

File: cli/cmd/tapRunner.go [hunk 2]
```diff
@@ -736,6 +739,89 @@ func watchTapperPod(ctx context.Context, kubernetesProvider *kubernetes.Provider
...
+			if event.Type == core.EventTypeWarning {
+				logger.Log.Warningf(uiUtils.Warning, fmt.Sprintf("Resource %s in state %s - %s", event.Regarding.Name, event.Reason, event.Note))
+			}
+		case err, ok := <-errorChan:
+			if !ok {
+				errorChan = nil
+				continue
+			}
+
+			logger.Log.Errorf("[Error] Error in mizu tapper watch, err: %v", err)
```
Comment (RoyUP9) [inline cli/cmd/tapRunner.go:815]: i think this should be error in mizu events or something like that

File: shared/kubernetes/watchEvent.go [hunk 3]
```diff
@@ -16,3 +18,12 @@ func (we *WatchEvent) ToPod() (*corev1.Pod, error) {
 
 	return pod, nil
 }
+
+func (we *WatchEvent) ToEvent() (*eventsv1.Event, error) {
+	pod, ok := we.Object.(*eventsv1.Event)
```
Comment (RoyUP9) [inline shared/kubernetes/watchEvent.go:23]: rename pod to event

File: shared/kubernetes/watchEvent.go [hunk 4]
```diff
@@ -16,3 +18,12 @@ func (we *WatchEvent) ToPod() (*corev1.Pod, error) {
...
+	if !ok {
+		return nil, fmt.Errorf("Invalid object type on pod event stream")
```
Comment (RoyUP9) [inline shared/kubernetes/watchEvent.go:25]: maybe change to something that is not similar to the error we get on ToPod

File: shared/kubernetes/watchEvent.go (hunk 3 above)
Comment (nimrod-up9) [reply shared/kubernetes/watchEvent.go]: Fixed

File: shared/kubernetes/watchEvent.go (hunk 4 above)
Comment (nimrod-up9) [reply shared/kubernetes/watchEvent.go]: done

File: cli/cmd/tapRunner.go (hunk 1 above)
Comment (nimrod-up9) [reply cli/cmd/tapRunner.go:758]: I'm thinking of removing the splitting of the event channel to added, removed, modified and error.
Instead, run a switch case when handling wEvent.
What do you think?

File: cli/cmd/tapRunner.go (hunk 2 above)
Comment (nimrod-up9) [reply cli/cmd/tapRunner.go]: done

File: cli/cmd/tapRunner.go (hunk 1 above)
Comment (RoyUP9) [reply cli/cmd/tapRunner.go:758]: i think it's a good idea, but maybe we should do it in separate PR

File: agent/main.go [hunk 5]
```diff
@@ -482,3 +485,86 @@ func startMizuTapperSyncer(ctx context.Context) (*kubernetes.MizuTapperSyncer, e
 
 	return tapperSyncer, nil
 }
+
+func watchMizuEvents(ctx context.Context, kubernetesProvider *kubernetes.Provider, cancel context.CancelFunc) {
```
Comment (RoyUP9) [inline agent/main.go:489]: isn't it the exact same function that we used for cli?

**Review by RoyUP9 — DISMISSED** [review]

**Review by RoyUP9 — APPROVED** [review]

**Review by RoyUP9 — APPROVED** [review]

**Pull request merged**